### PR TITLE
[chore] [exporter/alertmanagerexporter] Add exporter to components

### PR DESCRIPTION
**Pull request by mcube8**
**Description:** Add Alertmanager Exporter to Components
Third PR - Update buildconfig and components

**Link to tracking Issue:** [#23659](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/23569)

**Testing:** Unit tests for exporter

**Documentation:** Readme and Sample Configs to use Alertmanager exporter

### CODE REVIEW

**Review by songy23 — COMMENTED** [review]
`Replace statements are out of date, please run "make crosslink" and commit the changes in this PR.`

**Comment by mcube8** [comment]
> `Replace statements are out of date, please run "make crosslink" and commit the changes in this PR.`

Added a commit with the change, thank you.

**Review by songy23 — APPROVED** [review]

**Review by atoulme — APPROVED** [review]

**Comment by sokoide** [comment]
Hello @bryan-aguilar, the PR needs one more review/approval. We'd appreciate your help!

File: cmd/otelcontribcol/exporters_test.go
Comment (bryan-aguilar) [inline cmd/otelcontribcol/exporters_test.go:24]: Is there a reason that this requires a named import?

File: cmd/otelcontribcol/exporters_test.go
Comment (bryan-aguilar) [inline cmd/otelcontribcol/exporters_test.go:284]: This lifecycle test should be added to your `metadata.yaml` file instead of explicitly in the builder config. mdatagen will then automatically generate the configuration in the component. See[ the schema doc](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/4ad75b45ff83472040b9f2c4636237a2b4b9876d/cmd/mdatagen/metadata-schema.yaml#L109) for reference.

I'm going to mark this as a requirement to merge the PR so that we do not create more work for ourselves in the future. @atoulme is doing spearheading a lot of work to deprecate this test file by removing lifecycle tests to components.

Comment (bryan-aguilar) [reply cmd/otelcontribcol/exporters_test.go:284]: Issue ref https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/27849

**Review by bryan-aguilar — CHANGES_REQUESTED** [review]
Hi @sokoide,

For functional changes to PRs it is very helpful to have a codeowner approval on the PR. It gives collector approvers and maintainers, who may not be familiar with the component, more confidence in adding the `ready to merge` label on the PR. 

Would you mind also reviewing the PR and adding your approval, comment, or change request? 

Also, it looks like there are some merge conflicts that need to be resolved @mcube8

File: cmd/otelcontribcol/exporters_test.go
Comment (bryan-aguilar) [reply cmd/otelcontribcol/exporters_test.go:284]: Looks like this will be handled in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/30389/files

Comment (bryan-aguilar) [reply cmd/otelcontribcol/exporters_test.go:284]: #30389 is merged. This bit of code can be removed once you rebase from main.

**Review by bryan-aguilar — APPROVED** [review]

File: cmd/otelcontribcol/exporters_test.go
Comment (bryan-aguilar) [reply cmd/otelcontribcol/exporters_test.go:284]: @sokoide do you have the requisite permission to review PRs in the repository? I understand they won't be "green check mark" but even "grey check mark" helps. If you don't have permission for "grey check mark" review, then that may be an issue we need to raise.

Comment (mcube8) [reply cmd/otelcontribcol/exporters_test.go]: @bryan-aguilar Thanks for your review. I updated the test config in metadata.yml under alertmanager exporter component. When removing above snippet from exporters_test, the unittest check failed. 
Kindly suggest if any further changes are required, thanks!

**Pull request merged**
